### DISCUSSION TITLE
Deeplinking in StackNavigator

### DISCUSSION THREAD

**Discussion by damienherve** [react-navigation/react-navigation#10969] [Q&A] [answered]
Hi everyone,

i have a StackNavigator with 3 Screens: A / B / C
In the app, the flow classic flow is :
navigate('A') -> navigate('B') -> navigate('C') resulting in a StackNavigator with A -> B -> C in the end.

Is it possible to achieve this with a deeplinking, i mean show the screen C with A -> B loaded before it ?

When i read the doc, i can only load one screen in the stack, so just C, and eventually use initaiRouteName to load A or B before it (without params, which is another problem...).

Is there a way to load a StackNavigator with A -> B -> C from a deeplink ?

**Comment by satya164** [comment, marked as answer]
You can achieve that by specifying a custom `getStateFromPath` function where you return your desired state structure
